extensions/2.0/Khronos/KHR_displaymapping_pq/README.md

```diff
@@ -0,0 +1,439 @@
+If available, a framebuffer colorspace that is compatible with the color primaries of ITU BT.2020 shall be used.  
+If this colorspace is used the source color images must be converted from linear ITU BT.709 values to ITU BT.2020 linear values, where applicable depending on texture target and source format.  
+
+For HDR output,  a range extension value of 59.5208 and gamma of 2.4 shall be used.  
+
+**Implementation notes**
+
+Conversion of color primaries to ITU BT.2020 could be done after loading of a PNG/JPEG and after the image has been gamma expanded from sRGB to linear.  
+
+
+### SDR capable display
+
+If the framebuffer format or colorspace is not known, or none is available that preserves range, precision and color gamut then lower range framebuffer with sRGB colorspace may be used.  
+This is to allow for compatibility with displays that does not support higher range and/or compatible colorspaces.  
+It also allows for implementations where the details of the framebuffer is not known or available.  
+
+A SDR capable display is defined as having less than 10 bits per pixel for each colorchannel.  
+
+For SDR output,  a range extension value of 46.42 and gamma of 2.4 shall be used.  
+
+## Color source images
+
+If display type is considered to be HDR then color source images shall be converted to BT.2020 as needed.  
+
+
+### Color conversion matrix
+
+Color conversion from BT.709 to BT.2020 is specified in ITU BT.2087  
+https://www.itu.int/rec/R-REC-BT.2087/en
+
+The M2 linear color conversion matrix is defined as:  
+0.6274 0.3293 0.0433  
+0.0691 0.9195 0.0114  
+0.0164 0.0880 0.8956  
+
+
+## Scene aperture
+
+### Parameters
+
+The following parameters are added by the `KHR_displaymapping_pq` extension, it is added with the extension on a scene object:  
+
+| Name              | Type     | Description                                                                                                                                | Required |
+| ----------------- | -------- | ------------------------------------------------------------------------------------------------------------------------------------------ | -------- |
+| **sceneAperture** | `object` | Scene light adjustment setting, if no value supplied the default value of OFF will be used and scene light values will be kept unmodified. | No       |
+
+### Scene aperture
+
+`sceneAperture`
+
+| Name                | Type     | Description                                                                                          | Required |
+| ------------------- | -------- | ---------------------------------------------------------------------------------------------------- | -------- |
+| **apertureValue**   | `number` | Base value used to calculate aperture factor                                                         | No       |
```

Comment:
Since the JSON object containing this property is defined on an `aperture` object, the property should be called `value`.

Moreover, please consider moving all members of the `aperture` object to the extension object to reduce nesting.

Reply:
Thanks - updates

I would like to keep the aperture properties as an object that implementations may switch behavior depending on object declared or not.
Please re-open if you disagree.

Reply:
Let's agree on the intention first. The aperture object itself and all its four properties are optional, i.e., it's valid to omit them. Luminance range properties have default values. Here are two examples with their interpretation as written in the spec:

### Empty extension object
Pass scene light intact.
```json
{
  "scenes": [
    {
      "node": 0,
      "extensions": {
        "KHR_displaymapping_pq": {}
      }
    }
  ]
}
```

### Empty aperture object
Clamp scene luminance to `0.0 .. 10000.0`, **factor remains undefined**, result is undefined.
```json
{
  "scenes": [
    {
      "node": 0,
      "extensions": {
        "KHR_displaymapping_pq": {
          "aperture": {}
        }
      }
    }
  ]
}
```

Please elaborate on the expected usage of the aperture controls and I'll help with writing the correct schemas.

Reply:
Thanks for the offer @lexaknyazev :-)

 I think I understand your point know.
 
 The intention is that if `aperture` is not defined then no factoring of scene luminance will be done.
 The scene aperture step will be omitted - this is intended only for usecases where you know that your scene light is within range (0 - 10 000)
 
Do you mean that as it is specified now the scene aperture will _always_ be applied?

Reply:
> Do you mean that as it is specified now the scene aperture will _always_ be applied?

I mean that the current design allows defining an empty aperture object that would not work at all because `aperture.value` and `aperture.control` do not have default values. There are two possible solutions to this issue: make them required (my preference here) or define default values for them.

Reply:
> I mean that the current design allows defining an empty aperture object that would not work at all because `aperture.value` and `aperture.control` do not have default values.

Ah, I see - that is an unintentional error.
It should state that all values are required if the object is defined.
In the schema (gltf.aperture.schema) I have declared them as "minItems": 1

Is this enough (if I also change the table to say Yes)?

Reply:
> It should state that all values are required

As the min and max luminance properties have default values, they cannot be required.

> In the schema (gltf.aperture.schema) I have declared them as "minItems": 1

That rule refers to the length of arrays and cannot be used with numbers.
 
---

> Is this enough

Let's agree on the semantics in this thread first.

- Could any of these four properties be negative?
- Could `aperture.value` be zero?
- Could `maxLuminance` be less than `minLuminance`?

Reply:
Resolving this according to review meeting - please re-open if relevant after new changes